capstone/src/test/java/com/google/sps/clubs/ClubServletTest.java

```diff
@@ -74,61 +78,110 @@ public void tearDown() {
-                    request.getParameter(Constants.CLUB_NAME_PROP)));
+                    request.getParameter(Constants.PROPERTY_NAME)));
     Entity clubEntity = datastore.prepare(query).asSingleEntity();
 
-    Assert.assertEquals(SAMPLE_CLUB_NAME, clubEntity.getProperty(Constants.CLUB_NAME_PROP));
+    Assert.assertEquals(SAMPLE_CLUB_NAME, clubEntity.getProperty(Constants.PROPERTY_NAME));
     Assert.assertEquals(SAMPLE_CLUB_DESC_1, clubEntity.getProperty(Constants.DESCRIP_PROP));
     Assert.assertEquals(SAMPLE_CLUB_WEB, clubEntity.getProperty(Constants.WEBSITE_PROP));
     Assert.assertEquals(SAMPLE_BLOB, clubEntity.getProperty(Constants.LOGO_PROP));
     Assert.assertEquals(STUDENT_LIST, clubEntity.getProperty(Constants.MEMBER_PROP));
     Assert.assertEquals(STUDENT_LIST, clubEntity.getProperty(Constants.OFFICER_PROP));
 
-    Mockito.verify(response).sendRedirect(VALID_URL);
+    Mockito.verify(response).sendRedirect("/registration-msg.html?is-valid=true");
   }
 
   @Test
   public void doPost_registerNewInvalidClub() throws ServletException, IOException {
-    helper.setEnvEmail(TEST_EMAIL).setEnvAuthDomain("google.com").setEnvIsLoggedIn(true);
-    when(request.getParameter(Constants.CLUB_NAME_PROP)).thenReturn(SAMPLE_CLUB_NAME);
-    when(request.getParameter(Constants.DESCRIP_PROP)).thenReturn(SAMPLE_CLUB_DESC_1);
-    when(request.getParameter(Constants.WEBSITE_PROP)).thenReturn(SAMPLE_CLUB_WEB);
-
-    ImmutableList<BlobKey> keys = ImmutableList.of(SAMPLE_BLOB);
-    ImmutableMap<String, List<BlobKey>> blobMap = ImmutableMap.of(Constants.LOGO_PROP, keys);
-    when(blobstore.getUploads(request)).thenReturn(blobMap);
-
+    doPost_helper();
     clubServlet.doPostHelper(request, response, blobstore, datastore);
 
-    when(request.getParameter(Constants.DESCRIP_PROP)).thenReturn(SAMPLE_CLUB_DESC_2);
+    when(request.getParameter(Constants.DESCRIP_PROP)).thenReturn("club desc");
     clubServlet.doPostHelper(request, response, blobstore, datastore);
 
-    Mockito.verify(response).sendRedirect(INVALID_URL);
+    Mockito.verify(response).sendRedirect("/registration-msg.html?is-valid=false");
     Query query =
         new Query("Club")
             .setFilter(
                 new FilterPredicate(
-                    Constants.CLUB_NAME_PROP,
+                    Constants.PROPERTY_NAME,
                     FilterOperator.EQUAL,
-                    request.getParameter(Constants.CLUB_NAME_PROP)));
+                    request.getParameter(Constants.PROPERTY_NAME)));
     Entity clubEntity = datastore.prepare(query).asSingleEntity();
     Assert.assertEquals(SAMPLE_CLUB_DESC_1, clubEntity.getProperty(Constants.DESCRIP_PROP));
   }
+
+  @Test
+  public void doGet_clubExists() throws ServletException, IOException {
+    when(request.getParameter(Constants.PROPERTY_NAME)).thenReturn(SAMPLE_CLUB_NAME);
+    ImmutableList<String> expectedMembers =
+        ImmutableList.of("student@example.com", "officer@example.com");
```

Comment:
nit: I think anything that gets used twice in different places should get extracted into a local variable, just so you absolutely avoid accidentally getting errors due to typos here.